crates/extensions/c8y_mapper_ext/src/availability/actor.rs

```diff
@@ -0,0 +1,287 @@
+                    self.start_heartbeat_timer_if_interval_is_positive(source)
+                        .await?;
+                }
+                RegistrationResult::Update => {
+                    self.start_heartbeat_timer_if_interval_is_positive(source)
+                        .await?;
+                }
+                RegistrationResult::Error(reason) => warn!(reason),
+            },
+            EntityType::Service => {}
+        }
+
+        Ok(())
+    }
+
+    /// Insert a <"device topic ID" - "health entity topic ID" and "external ID"> pair into the map.
+    /// If @health is provided in the registration message, use the value as long as it's valid as entity topic ID.
+    /// If @health is not provided, use the "tedge-agent" service topic ID as default.
+    /// @id is the only source to know the device's external ID. Hence, @id must be provided in the registration message.
+    fn update_device_service_pair(
+        &mut self,
+        registration_message: &EntityRegistrationMessage,
+    ) -> RegistrationResult {
+        let source = &registration_message.topic_id;
+
+        let result = match registration_message.other.get("@health") {
+            None => Ok(registration_message
+                .topic_id
+                .default_service_for_device("tedge-agent")
+                .unwrap()),
+            Some(raw_value) => match raw_value.as_str() {
+                None => Err(format!("'@health' must hold a string value. Given: {raw_value:?}")),
+                Some(maybe_entity_topic_id) => EntityTopicId::from_str(maybe_entity_topic_id)
+                    .map_err(|_| format!("'@health' must be 4-segment identifier like `a/b/c/d`. Given: {maybe_entity_topic_id}"))
+            }
+        };
+
+        match result {
+            Ok(health_topic_id) => {
+                match registration_message.external_id.clone() {
+                    None => RegistrationResult::Error(format!("'@id' field is missing. Cannot start availability monitoring for the device '{source}'")),
+                    Some(external_id) => {
+                        match self.device_ids_map
+                            .insert(source.clone(), DeviceIds { health_topic_id, external_id }) {
+                            None => RegistrationResult::New,
+                            Some(_) => RegistrationResult::Update,
+                        }
+                    }
+                }
+            }
+            Err(err) => RegistrationResult::Error(format!("'@health' contains invalid value in {source}. Details: {err}")),
+        }
+    }
+
+    /// Set a new timer for heartbeat if the given interval is positive value
+    async fn start_heartbeat_timer_if_interval_is_positive(
+        &mut self,
+        source: &EntityTopicId,
+    ) -> Result<(), RuntimeError> {
+        if !self.config.interval.is_zero() {
+            self.timer_sender
+                .send(SetTimeout::new(
+                    self.config.interval,
```

Comment:
```suggestion
                    self.config.interval / 2,
```
Reason explained in [this comment](https://github.com/thin-edge/thin-edge.io/pull/2866#discussion_r1611209242) from the old PR.

Reply:
Done 4fea7a84dd83af120f026fef0771eea262013c17